### ISSUE TITLE
PRO: Stable cursor ignores Arel orderings using default direction

### ISSUE THREAD

**Issue by kobsy** [rmosolgo/graphql-ruby#3605]
**graphql version:** 1.12.5  
**graphql-pro version:** 1.17.8

When working with the Postgres stable pagination cursors, I noticed some records were being skipped when attempting to fetch all pages. Digging in, it looks like the feature uses the orderings present on a relation to build the conditions for fetching pages. However, it only seems to recognize `Arel::Nodes::Descending`, `Arel::Nodes::Ascending`, and `String` order values, and any values that aren't of these types are ignored when building the cursors and finding pages. This ends up omitting any values that reference an Arel column without specifying a direction. So by way of example:

```ruby
# Good. Just uses ActiveRecord to order by a column, which
# is transformed to an Arel::Nodes::Ascending under-the-hood
Model.order(:position)

# Bad. Naked Arel column is ignored and only models.position is used!
Model.joins(:category).order(Category.arel_table[:position], :position)

# Good. Things work again once we're explicit about direction
Model.joins(:category).order(Category.arel_table[:position].asc, :position)
```

As demonstrated above, there's a straightforward workaround. That said, I found it surprising that the Arel column ordering was ignored -- and silently at that! I wonder if the ordering extractor could add recognition for order values that respond to `to_sql`? 🤔 Or failing that, perhaps emit a warning when there are order values that end up being ignored (especially since there's such a straightforward workaround)? Or even including in the documentation that a direction has to be supplied for stable cursor support to work? The hope is that others could avoid the bullet we dodged: having records dropped silently because of this particular quirk. 🙂

Thanks for all your hard work on this invaluable set of gems! 😁

**Comment by rmosolgo** [comment]
Thanks for the detailed writeup! I agree that the arrangement you described above should work. (Admittedly, it's the first time I've seen it 🙈 !) 

> and silently at that!

Yes, I'll change that. I wonder what I was thinking 😖 

I'll take a look and follow up here soon!

**Comment by rmosolgo** [comment]
Just released this in graphql-pro 1.18.3. Let me know if you run into any trouble after upgrading, and thanks again for the great bug report!
